pkg/mock/fileUtils_test.go

```diff
@@ -288,3 +289,34 @@ func TestStat(t *testing.T) {
 		}
 	})
 }
+
+func TestGetChod(t *testing.T) {
+	files := FilesMock{}
+	files.AddDirWithMode("tmp", 0777)
+	files.AddFileWithMode("tmp/log.txt", []byte("build failed"), 0777)
+
+	t.Run("non existing file", func(t *testing.T) {
+		err := files.Chmod("does/not/exist", 0400)
+		assert.EqualError(t, err, "chmod: does/not/exist: No such file or directory")
+	})
+
+	t.Run("chmod for file", func(t *testing.T) {
+		err := files.Chmod("tmp/log.txt", 0644)
+		if assert.NoError(t, err) {
+			info, e := files.Stat("tmp/log.txt")
+			if assert.NoError(t, e) {
+				assert.Equal(t, othersCanRead, info.Mode())
+			}
+		}
+	})
+
+	t.Run("chmod for directory", func(t *testing.T) {
+		err := files.Chmod("tmp", 0755)
+		if assert.NoError(t, err) {
+			info, e := files.Stat("tmp")
+			if assert.NoError(t, e) {
+				assert.Equal(t, othersCanReadAndExecute, info.Mode())
+			}
+		}
+	})
```

Comment:
The way you have written the test actually reveals a short-comming in the Chmod() implementation. Short version: The test should also work without the line `files.AddDirWithMode("tmp", 0777)` in the setup phase. Long version: A directory is also considered to exist, if a file has been added with a path. But at this point, no properties will exist for that folder. Chmod() should detect this situation and add an entry in the path map just for storing the properties. What do you think? (Actually, deleting a file could be considered to have the same problem: the directory where it resided should still exist.)

Reply:
> reveals a short-comming in the Chmod() 

This is not a short-comming in the Chmod() impl (clarification: we are talking about the impl in the mock).

> A directory is also considered to exist, if a file has been added with a path

This is actually more a short-comming in the mock "as is". When we add a file to the mock with some nested path the path should be added at that point in time implicitly. When a file is removed, the directory structure should remain.

We can - of course - enhance the mock. But with mocking you will frequently have a bias between the mock and the real world impl. We have to keep a balance between being close to reality and efforts for getting this. In case it really hurts, we should adjust closer to reality IMO, if not I can life with the bias (... for now).

Reply:
Yes, this is of course about the mock implementation only. I understand your reasoning. The reason why I would be in favor of having a slightly more sophisticated implementation is only for the sake of self-consistency. We have DirExists(), which returns true also for implicitly created directories, so my reasoning would be that Chmod() on such dirs should work as well. That being said, I am fine with approving the PR as is, and improving this later, offering to do it myself.